Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -820,6 +861,79 @@ public Task Sort( DataGridColumn<TItem> column, SortDirection? sortDirection = n
         return Task.CompletedTask;
     }
 
+
+    /// <summary>
+    /// Replaces the current sorting with the given sort order
+    /// </summary>
+    /// <param name="columns">Columns used for sorting</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    /// <remarks>
+    /// Note that <see cref="DataGridColumn{TItem}.Sortable"/> and <see cref="Sortable"/> must be enabled to be able to sort!
+    /// If more than one column is specified, <see cref="SortMode"/> must be <see cref="DataGridSortMode.Multiple"/>
+    /// </remarks>
+    public async Task ApplySorting(params DataGridSortColumn[] columns ) {
+        if ( columns.Length > 1 && SortMode == DataGridSortMode.Single )
+        {
+            throw new ArgumentException(
+                $"Cannot sort on multiple columns when {nameof(SortMode)} is {nameof(DataGridSortMode.Single)}." );
+        }
+
+        if ( SortMode == DataGridSortMode.Single )
+        {
+            if ( columns.Length == 1 )
+            {
+                await Sort( columns[0].Field, columns[0].SortDirection );
+            }
+            else if ( SortByColumns.Count == 1 )
+            {
+                await Sort( SortByColumns[0].Field, SortDirection.Default );
+            }
+
+            return;
+        }
+
+        // reset sorting
+        foreach ( var column in SortByColumns )
+        {
+            column.CurrentSortDirection = SortDirection.Default;
+            await column.ResetSortOrder();
+        }
+
+        SortByColumns.Clear();
+
+        // apply new sorting
+        var columnTuples = columns
+            .Select( ( x, idx ) => (
+                Column: Columns.SingleOrDefault( c => c.Field == x.Field ),

```

Comment:
Why **SingleOrDefault**? User could define two columns with same Field for some reason. Maybe better not throw in that case.

Reply:
I agree that we should use only `FirstOrDefault`.